kyuubi-main/src/test/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilderSuite.scala

```diff
@@ -179,6 +180,27 @@ class SparkProcessBuilderSuite extends KerberizedTestHelper {
       atomicTest()
     }
   }
+
+  test("overwrite log file should cleanup before write") {
+    val fakeWorkDir = Files.createTempDirectory("fake")
+    val builder1 = new FakeSparkProcessBuilder(KyuubiConf()) {
+      override val workingDir: Path = fakeWorkDir
+    }
+    val file1 = builder1.engineLog
+    Files.write(file1.toPath, "a".getBytes(), StandardOpenOption.APPEND)
+    assert(file1.length() == 1)
+    Files.write(file1.toPath, "a".getBytes(), StandardOpenOption.APPEND)
+    assert(file1.length() == 2)
+    file1.setLastModified(System.currentTimeMillis() - Duration.ofDays(1).toMillis - 1000)
```

Comment:
Should we explicitly set ENGINE_LOG_TIMEOUT rather than depends on default value?

Reply:
Yeah, use the config exsplicitly.